packages/core/src/virtual-content.ts

```diff
@@ -10,29 +11,40 @@ const externalCJSPrefix = '/@embroider/ext-cjs/';
 // this produces the corresponding contents. It's a static, stateless function
 // because we recognize that that process that did resolution might not be the
 // same one that loads the content.
-export function virtualContent(filename: string, resolver: Resolver): string {
+export function virtualContent(filename: string, resolver: Resolver): { src: string; watches: string[] } {
+  if (filename === '/the-file-list') {
+    let dir = join(resolver.options.appRoot, 'app');
+    let files = readdirSync(dir);
+    return {
+      src: `export default [
+      ${files.map(f => `"${f}"`).join(',')}
```

Comment:
this specific case could probably also be solved by using vite `import.meta.glob('./dir/*')` . would it sill not do auto reload in that case?
other cases could solve it by creating imports to files, vite also supports `?raw` which would allow to include any file that needs to be in a virtual file.